src/main/java/org/fogbowcloud/manager/core/threads/OpenProcessor.java

```diff
@@ -0,0 +1,177 @@
+	 */
+	protected void processOpenOrder(Order order) {
+		synchronized (order) {
+			OrderState orderState = order.getOrderState();
+
+			if (orderState.equals(OrderState.OPEN)) {
+				LOGGER.info("Trying to get an Instance for Order [" + order.getId() + "]");
+
+				try {
+					InstanceProvider instanceProvider = this.getInstanceProviderForOrder(order);
+
+					LOGGER.info("Processing Order [" + order.getId() + "]");
+					order.handleOpenOrder();
+					OrderInstance orderInstance = instanceProvider.requestInstance(order);
+					order.setOrderInstance(orderInstance);
+
+					LOGGER.info("Removing Order [" + order.getId() + "] from Open Orders List");
+					this.openOrdersList.removeItem(order);
+
+					LOGGER.info("Updating Order State after processing [" + order.getId() + "]");
+					this.updateOrderStateAfterProcessing(order);
+
+				} catch (Exception e) {
+					LOGGER.error("Error while trying to get an Instance for Order: " + System.lineSeparator() + order,
+							e);
+					order.setOrderState(OrderState.FAILED);
+
+					LOGGER.info("Adding Order [" + order.getId() + "] to Failed Orders List");
+					this.failedOrdersList.addItem(order);
+				}
+			}
+		}
+	}
+
+	/**
+	 * After processing an Open Order, is necessary update it state.
+	 * 
+	 * @param order
+	 */
+	protected void updateOrderStateAfterProcessing(Order order) {
+		if (order.isLocal(this.localMemberId)) {
+			OrderInstance orderInstance = order.getOrderInstance();
+			String orderInstanceId = orderInstance.getId();
+
+			if (!orderInstanceId.isEmpty()) {
+				LOGGER.info("The open order [" + order.getId() + "] got an local instance with id [" + orderInstanceId
+						+ "], setting your state to SPAWNING");
+
+				order.setOrderState(OrderState.SPAWNING);
+
+				LOGGER.info("Adding Order [" + order.getId() + "] to Spawning Orders List");
+				this.spawningOrdersList.addItem(order);
+
+			} else {
+				LOGGER.error("Order Instance Id for Order [" + order.getId() + "] is Empty");
+				throw new RuntimeException("Order Instance Id for Order [" + order.getId() + "] is Empty");
+			}
+
+		} else if (order.isRemote(this.localMemberId)) {
+			LOGGER.info("The open order [" + order.getId()
+					+ "] was requested for remote member, setting your state to PENDING");
+
+			order.setOrderState(OrderState.PENDING);
+
+			LOGGER.info("Adding Order [" + order.getId() + "] to Pending Orders List");
+			this.pendingOrdersList.addItem(order);
+		}
+	}
+
+	/**
+	 * Get the Instance Provider for an Order, if the Order is Local, the
+	 * returned Instance Provider is the Local, else, is the Remote.
+	 * 
+	 * @param order
+	 * @return
+	 */
+	protected InstanceProvider getInstanceProviderForOrder(Order order) {
+		InstanceProvider instanceProvider = null;
+		if (order.isLocal(this.localMemberId)) {
+			LOGGER.info("The open order [" + order.getId() + "] is local");
+
+			instanceProvider = this.localInstanceProvider;
+		} else if (order.isRemote(this.localMemberId)) {
```

Comment:
Would it be an else statement?

Reply:
Done, [here](https://github.com/fogbow/fogbow-manager-core/pull/13/commits/71e95e679ef7a149f19ffe48e1e4c50ba4fdfd0d).